quickbooks/client.py

```diff
@@ -175,14 +198,19 @@ def make_request(self, request_type, url, request_body=None, content_type='appli
         }
 
         req = self.session.request(request_type, url, True, self.company_id, headers=headers, params=params, data=request_body)
+        if req.status_code == httplib.UNAUTHORIZED:
+            raise AuthorizationException("Application authentication failed")
```

Comment:
It would be really helpful here to add more information to the exception to help distinguish between different cases : 

`raise AuthorizationException("Application authentication failed", detail=req.text)`

And then update the exception to display something like:

```
class AuthorizationException(QuickbooksException):
    def __str__(self):
        return "QB Auth Exception: " + self.message + " \n\n" + self.detail
```

Currently, when I get this error I have no idea what exactly failed with authorization, which makes it really hard to debug.